File: HappyCollisions/Actors/PointActor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace HappyCollisions.Actors
{
    class PointActor : IActor
    {
        private double x;
        private double y;
        private double vx;
        private double vy;

        public double X { get => x; set => x = value; }
        public double Y { get => y; set => y = value; }
        public double VX { get => vx; set => vx = value; }
        public double VY { get => vy; set => vy = value; }

        public PointActor(double x, double y, double vx, double vy)
        {
            this.x = x;
            this.y = y;
            this.vx = vx;
            this.vy = vy;
        }
    }
}

File: CollisionFun/Actors/PointActor.fs
module Actors.PointActor

open OpenTK
open Interfaces

type public PointActor (position : Vector2d, velocity : Vector2d) =
    let mutable position : Vector2d = position
    let mutable velocity : Vector2d = velocity
    interface IActor with
        member __.Position: Vector2d = 
            position
        member __.Velocity: Vector2d = 
            velocity
        member __.Accelerate(a: Vector2d): unit = 
            velocity <- velocity + a
        member __.Move (delta : Vector2d) : unit =
            position <- position + delta